home/tests/test_api.py

```diff
@@ -244,118 +254,142 @@ def test_pagination(self):
+        """
+        FIXME:
+         * Is the number of queries significant? What does it depend on?
+         * Should this be part of some other test?
+        """
+        with django_assert_num_queries(8):
+            uclient.get("/api/v2/pages/")
 
-    def test_detail_view(self):
+    def test_detail_view(self, uclient):
+        """
+        FIXME:
+         * It's unclear what this is actually testing.
+         * Should it be multiple tests instead of just one?
+         * This should probably be in a class without the content setup.
+        """
         ContentPage.objects.all().delete()
-        self.assertEqual(PageView.objects.count(), 0)
+        assert PageView.objects.count() == 0
 
         page = create_page(tags=["tag1", "tag2"])
 
         # it should return the correct details
-        response = self.client.get(f"/api/v2/pages/{page.id}/")
+        response = uclient.get(f"/api/v2/pages/{page.id}/")
         content = response.json()
 
-        self.assertEqual(content["id"], page.id)
-        self.assertEqual(content["title"], page.title)
-        self.assertEqual(content["tags"], ["tag1", "tag2"])
-        self.assertFalse(content["has_children"])
+        assert content["id"] == page.id
+        assert content["title"] == page.title
+        assert content["tags"], ["tag1" == "tag2"]
+        assert not content["has_children"]
 
-        self.assertEqual(PageView.objects.count(), 1)
+        assert PageView.objects.count() == 1
 
         # if there are children pages
         create_page("child page", page.title)
 
-        response = self.client.get(f"/api/v2/pages/{page.id}/?whatsapp=True")
+        response = uclient.get(f"/api/v2/pages/{page.id}/?whatsapp=True")
         content = response.json()
 
-        self.assertTrue(content["has_children"])
+        assert content["has_children"]
 
-        self.assertEqual(PageView.objects.count(), 2)
+        assert PageView.objects.count() == 2
         view = PageView.objects.last()
-        self.assertEqual(view.message, None)
+        assert view.message is None
 
         # if we select the whatsapp content
-        response = self.client.get(f"/api/v2/pages/{page.id}/?whatsapp=true&message=1")
+        response = uclient.get(f"/api/v2/pages/{page.id}/?whatsapp=true&message=1")
         content = response.json()
 
-        self.assertEqual(content["title"], page.whatsapp_title)
+        assert content["title"] == page.whatsapp_title
 
-        self.assertEqual(PageView.objects.count(), 3)
+        assert PageView.objects.count() == 3
         view = PageView.objects.last()
-        self.assertEqual(view.message, 1)
+        assert view.message == 1
 
-    def test_detail_view_with_variations(self):
+    def test_detail_view_with_variations(self, uclient):
+        """
+        FIXME:
+         * It's unclear what this is actually testing.
+         * This should probably be in a class without the content setup.
+        """
```

Comment:
This is intended to test the `variation_messages` key in the API result.

Reply:
Since variations are (currently) WhatsApp-only, should I move this to `TestWhatsAppMessages`?

Reply:
Done.